<!-- 
  Works with rocon_hub/hub.launch. 
-->
<launch>
  <arg name="gateway_name" default="gateway"/>
  <arg name="gateway_watch_loop_period" default="5"/>
  <arg name="gateway_hub_whitelist" default="[]"/> <!-- list of hub names/regex patterns -->
  <arg name="gateway_network_interface" default="$(optenv GATEWAY_NETWORK_INTERFACE)"/>  <!-- If you have multiple network interfaces and want to lock it on one (e.g. 'eth0') -->
  <arg name="gateway_disable_uuids" default="false"/> <!-- manage unique naming of multiple gateways yourself -->
  <arg name="gateway_firewall" default="true"/> <!-- Typically nothing comes in! -->

  <node ns="zeroconf" pkg="zeroconf_avahi" type="zeroconf" name="zeroconf"/>
  <node pkg="rocon_gateway" type="gateway.py" name="gateway" output="screen">
    <rosparam command="load" file="$(find rocon_gateway)/param/default.yaml" />
    <rosparam command="load" file="$(find rocon_gateway)/param/default_blacklist.yaml" />
    <rosparam param="hub_whitelist" subst_value="True">$(arg gateway_hub_whitelist)</rosparam>
    <param name="name" value="$(arg gateway_name)"/>
    <param name="watch_loop_period" value="$(arg gateway_watch_loop_period)"/>
    <param name="network_interface" value="$(arg gateway_network_interface)"/>
    <param name="firewall" value="$(arg gateway_firewall)"/>
    <param name="disable_uuids" value="$(arg gateway_disable_uuids)"/>
  </node>
</launch>
